src/Database/Migrations/MigrationManager.php

```diff
@@ -63,4 +93,24 @@ public function executeUp(MigrationInterface $migration): void
 
         event(new MigrationMigrated($migration->getName()));
     }
+
+    public function executeDown(MigrationInterface $migration): void
+    {
+        $query = $migration->down();
+
+        if (! $query) {
+            return;
+        }
+
+        $this->database->execute($query);
+
+        try {
+            Migration::firstWhere('name', Operator::Equals, $migration->getName())
+                ->delete();
+        } catch (PDOException) {
+            // @todo should be handled better as PDO exception doesn't necessarily mean that the migrations table doesn't exist
```

Comment:
Also dispatch an event so that the user knows a migration didn't run. Also: shouldn't we stop migrations as soon as an error occurs?

Reply:
I've added a comment in the code to better explain this case, lemme know if it makes sense!